### PR TITLE
Add ReleaseHowTo document

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
Add the document to record the release steps.

### CODE REVIEW

File: docs/ReleaseHowTo.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+### How to release a new version of harvester CSI driver
+
+#### 1. Tag harvester CSI driver if needed. Sometimes, we could skip this part if only the support version was update on Rancher
+
+#### 2. Bump new version for Rancher apps
```
Comment (connorkuehl) [inline docs/ReleaseHowTo.md:5]: Would it be worthwhile to add a little bit more context here for brand-new Harvester devs?

> harvester-csi-driver is distributed as a [standalone Helm package](https://github.com/harvester/charts/tree/master/charts/harvester-csi-driver) and is included in [Rancher's App catalogue](https://github.com/harvester/charts/tree/master/charts/harvester-csi-driver).

File: docs/ReleaseHowTo.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  - upstream-version: means the upstream version (The version of this app).
+  - appVersion: means the real version of the harvester csi driver (refer to release version).
+  - packages/harvester/harvester-csi-driver/package.yaml - update the version. you could refer [here](https://github.com/rancher/charts#versioning-charts) to know how to fill it correctly.
+  - release.yaml - add the new version to the release.yaml file.
+
+    2.2. After updating the above files, you need to execute `make prepare`, `make patch` and `make clean`. Remember you could add variable PACKAGE=`harvester/harvester-csi-driver` to speed up this process.
+    
+    2.3. Commit the changes until now.
+    
+    2.4. user `PACKAGE=harvester/harvester-csi-driver make charts` to generate the new charts files.
```
Comment (connorkuehl) [inline docs/ReleaseHowTo.md:20]:
```suggestion
    2.4. Use `PACKAGE=harvester/harvester-csi-driver make charts` to generate the new charts files.
```

File: docs/ReleaseHowTo.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+### How to release a new version of harvester CSI driver
+
+#### 1. Tag harvester CSI driver if needed. Sometimes, we could skip this part if only the support version was update on Rancher
+
+#### 2. Bump new version for Rancher apps
+
+  2.1. For Rancher apps, you need to update the following files:
```
Comment (ibrokethecloud) [inline docs/ReleaseHowTo.md:7]: can we please clarify that this change needs to be made in the `rancher/charts` repo. As a first time user even I was a bit confused how changes in harvester-csi-driver repo were being replicated to the `rancher/charts` repo.

File: docs/ReleaseHowTo.md (hunk 1 above)
Comment (Vicente-Cheng) [reply docs/ReleaseHowTo.md:5]: Nice suggestion, updated!

File: docs/ReleaseHowTo.md (hunk 3 above)
Comment (Vicente-Cheng) [reply docs/ReleaseHowTo.md:11]: Updated! Please help to recheck!

File: docs/ReleaseHowTo.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+
+#### 4. Bump new version for Rancher rke2
+
+   4.1 Update the following files:
+   - Dockerfile - you can find the `CHART_VERSION` of harvester-csi-driver and update its latest version. (Please make sure the version is consistent with the above)
+   - scripts/build-images - fill the harvester-csi-driver release version for build images.
+
+   4.2 Reference PR:
+   - [Bump harvester csi driver to v0.1.16](https://github.com/rancher/rke2/pull/3999)
```
Comment (bk201) [inline docs/ReleaseHowTo.md:49]: The reference PR has a version `0.1.1600`, can we add some document about why.

File: docs/ReleaseHowTo.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  2.4. Use `PACKAGE=harvester/harvester-csi-driver make charts` to generate the new charts files.
+  
+  2.5 Commit these changes for the new chart.
+  
+  2.6 Run `make validate` to make sure the CI would be passed.
+
+  2.7 Reference PR:
+  - [Bump harvester-csi-driver to v0.1.16 and harvester-cloud-provider to v0.1.14](https://github.com/rancher/charts/pull/2450)
+
+#### 3. Bump new version for Rancher rke2 chart
```
Comment (bk201) [inline docs/ReleaseHowTo.md:33]:
```suggestion
#### 3. Bump new version for Rancher RKE2 chart
```
suggest using the upper case for the product name.

**Review by bk201 — COMMENTED** [review]
The document describe updating the csi charts in two place
- Rancher market
- RKE2 build-in charts
Can we split the prodecudes to two parts and add a bit more overview why we need to put charts there. Thanks!

File: docs/ReleaseHowTo.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
+## How to release a new version of harvester CSI driver
+
+#### 1. Tag harvester CSI driver if needed. Sometimes, we could skip this part if only the support version was update on Rancher
+
+#### 2. Bump new version for Rancher apps
+
+  Harvester CSI driver provides a standard CSI interface used by guest Kubernetes clusters on Rancher with Harvester.
+  It is a [standalone helm chart package](https://github.com/rancher/charts/tree/dev-v2.8/charts/harvester-csi-driver). We could install it on Rancher app directly.
```
Comment (ibrokethecloud) [inline docs/ReleaseHowTo.md:8]:
```suggestion
  It is a [standalone helm chart package](https://github.com/rancher/charts/tree/dev-v2.8/charts/harvester-csi-driver). We could install it as a Rancher app directly.
```

**Comment by ibrokethecloud** [comment]
just a few minor changes rest looks good.

File: docs/ReleaseHowTo.md (hunk 6 above)
Comment (Vicente-Cheng) [reply docs/ReleaseHowTo.md]: updated, thanks!

File: docs/ReleaseHowTo.md (hunk 5 above)
Comment (Vicente-Cheng) [reply docs/ReleaseHowTo.md]: Updated!

File: docs/ReleaseHowTo.md (hunk 4 above)
Comment (Vicente-Cheng) [reply docs/ReleaseHowTo.md:58]: Add a related description for it.

**Review by bk201 — APPROVED** [review]
LGTM, thanks!

**Review by connorkuehl — APPROVED** [review]

**Review by ibrokethecloud — APPROVED** [review]
LGTM. thanks.

**Review by markhillgit — APPROVED** [review]
good, thank you!

**Pull request merged**